deltacat/compute/compactor/utils/primary_key_index.py

```diff
@@ -8,10 +8,10 @@
 
 from deltacat.utils.common import ReadKwargsProvider
 from ray import cloudpickle
+from deltacat.constants import PRIMARY_KEY_INDEX_WRITE_BOTO3_CONFIG
```

Comment:
We may want to capture a TODO here that also helps decouple us from S3 in general by just writing the primary key index files back to the catalog using the existing DeltaCAT client APIs. This should defer all writes to the underlying file system in use by the catalog, which should already have considered strategies to better avoid/mitigate throttling issues from cloud storage providers like S3.

Reply:
Makes sense.